Set up meson project

notpeelz opened the pull request:
This should make it easier to refactor (#16).

You can try it out like this:

```
meson setup --prefix=/usr/local build
meson install -C build
```

Vladimir-csp commented:
It's funny to think that a bit over a year ago I naively thought this would be a simple shell script...

Thanks! I'll investigate everything.

notpeelz commented:
I'll be committing changes as `fixup!` commits instead of amending my existing commits. This is so that it's easier for you to understand what changes have been made at every step of the review.

Once I get your approval for merging, I'll rebase and squash the `fixup!` commits to clean up the history.
Just make sure not to merge my branch until then, because [GitHub doesn't automatically squash these special commits](https://github.com/orgs/community/discussions/40804).

notpeelz commented:
I removed the vscode-specific pyright/pylance config and replaced it with an editor-neutral `pyproject.toml` config: https://github.com/microsoft/pyright/blob/main/docs/configuration.md#sample-pyprojecttoml-file

Vladimir-csp commented on uwsm-plugins/meson.build:
```diff
@@ -0,0 +1,11 @@
+
+install_data(
+  uwsm_plugins,
+  install_dir: PKG_DATA_DIR / 'plugins',
```
Why move plugins away from lib?

notpeelz replied:
`/usr/share` seems to be the preferred location for sourced scripts.
There's a handful of packages that use `/usr/lib`, so it doesn't seem to be a hard rule.

e.g
bash: `/usr/share/bash-completion`
zsh : `/usr/share/zsh/functions`
nano: `/usr/share/nano/*.nanorc`
mpv: `/usr/share/mpv/scripts/*.sh`
etc

obs-studio is an interesting one. Because their plugins are in C/C++, the shared objects are stored in `/usr/lib/obs-plugins/`, but their runtime data is put in `/usr/share/obs/obs-plugins/<plugin-name>/`.
e.g the (3rd party) obs-vkcapture plugin:
```
/usr/lib/obs-plugins/linux-vkcapture.so
/usr/share/obs/obs-plugins/linux-vkcapture/
/usr/share/obs/obs-plugins/linux-vkcapture/locale/
/usr/share/obs/obs-plugins/linux-vkcapture/locale/cs-CZ.ini
/usr/share/obs/obs-plugins/linux-vkcapture/locale/de-DE.ini
/usr/share/obs/obs-plugins/linux-vkcapture/locale/en-US.ini
/usr/share/obs/obs-plugins/linux-vkcapture/locale/es-ES.ini
/usr/share/obs/obs-plugins/linux-vkcapture/locale/fr-FR.ini
/usr/share/obs/obs-plugins/linux-vkcapture/locale/it-IT.ini
/usr/share/obs/obs-plugins/linux-vkcapture/locale/ja-JP.ini
/usr/share/obs/obs-plugins/linux-vkcapture/locale/pl-PL.ini
/usr/share/obs/obs-plugins/linux-vkcapture/locale/pt-BR.ini
/usr/share/obs/obs-plugins/linux-vkcapture/locale/ru-RU.ini
/usr/share/obs/obs-plugins/linux-vkcapture/locale/uk-UA.ini
```

More on the topic: https://bbs.archlinux.org/viewtopic.php?pid=2049023

Vladimir-csp commented:
Let's go ahead with this. It will warrant bumping version up to 0.14 and a `[Breaking]` note about everything, including plugin paths.

notpeelz commented:
Great!

Should I add a new commit to bump the version and write about the breaking changes there?
BTW, have you considered using something like [conventional commits](https://www.conventionalcommits.org/en/v1.0.0)? The "breaking changes" part is signaled by the exclamation mark, e.g `chore!:`, `fix!:`, etc

Vladimir-csp commented:
> Should I add a new commit to bump the version and write about the breaking changes there?

Yes.

> BTW, have you considered using something like [conventional commits](https://www.conventionalcommits.org/en/v1.0.0)?

Good idea!

notpeelz commented:
I don't want to push my luck, but would you be open to adopting the [semantic versioning](https://semver.org) version format? It would make sense since you're already using a 3-component version format for non-zero minor releases.
e.g 0.14.0 instead of 0.14

Vladimir-csp commented:
No problem )

notpeelz commented:
I didn't mark the version bump commit as breaking, since it's technically not a breaking change itself. The breaking commits have `chore!`, `fix!`, etc.
I hope that's OK :)

Vladimir-csp commented on README.md:
```diff
@@ -6,12 +6,6 @@ wrapping standalone Wayland compositors into a set of systemd units.
 WIP(ish). The main structure of subcommands and features is more or less settled and will likely
 not receive any drastic changes unless some illuminative idea comes by.
 Nonetheless, keep an eye for commits with `[Breaking]` messages.
```
Could you also please change this, now that we are changing style of commits?

notpeelz replied:
![image](https://github.com/Vladimir-csp/uwsm/assets/920910/48fd095a-5266-4b05-827d-284540cff573)
Here's a preview of the README changes.
I've never used these icons, but they look pretty cool IMO.
https://github.com/orgs/community/discussions/16925

Vladimir-csp commented on README.md:
```diff
@@ -3,14 +3,21 @@
+> This simplifies environment management and allows proper cleanup.
+> The separate activation environment of the reference D-Bus implementation doesn't allow unsetting vars, so they're
+> set to an empty string instead, as a best effort cleanup. The only way to properly clean up in this case is
+> to run `loginctl terminate-user ...`.
```
Empty arg is a feature of loginctl, points to calling user.

notpeelz replied:
Whoops. TIL

Vladimir-csp commented:
Congrats!

notpeelz commented:
Thank you! I appreciated our conversation.
